src/config/create-jest-preset.ts

```diff
@@ -1,11 +1,14 @@
-import * as jestConfig from 'jest-config'
+import * as jestConfigType from 'jest-config'
 
 import { rootLogger } from '../util/logger'
 
+import { getJestConfigPkg } from './jest-config-resolver'
+
 const logger = rootLogger.child({ namespace: 'jest-preset' })
 
+const jestConfigPkg: typeof jestConfigType = getJestConfigPkg(logger)
```

Comment:
We use to start a type with an uppercase, and prefix with a T when it's a package export, so here I guess it should be named `TJestConfigPkg`

Anyway, here the getJestConfigPkg should be defined in a way so that the returned type is inferred here (I yet did not look at it in the jest-config-resolver-file). Long story short, you shouldn't have to `import * as jestConfigType` and here should only be `const jestConfigPkg = getJestConfigPkg(logger)`. Type will be inferred from function.

Reply:
Ok, just saw now, the type IS returned from `getJestConfig`, so simply remove the import and replace that line with what I wrote you.

Reply:
The type isn't implied because the `getJestConfig` is generic. I'll change it to not be generic and use the `TJestConfigPkg` name you suggested.

Reply:
Actually I remember why I did this the way I did it - giving `getJestConfigPkg` a non-generic return type and returning the real type (`typeof TJestConfigPkg`), the `jest-config-resolver.d.ts` declaration file takes a dependency on `jest-config`, which means that `ts-jest` can't be referenced by other TypeScript projects that don't have a flattened `node_modules`. I'm going to put it back the way I had it before to avoid this issue if that's okay with you.

Reply:
Oh yeah, good point! Tho if the type you are using is just defined for this file, move it out of types.ts and into this file.